### PR TITLE
Refactoring and fix for e2e acks

### PR DESCRIPTION
**Pull request by melzareix**
### Description
- Fix E2E Acknowledgements 
- Continuously poll Neptune Stream when `StreamRecordsNotFoundException` is encountered.
- Refactor some file names and classes
- Fixed some tests

### CODE REVIEW

File: data-prepper-plugins/neptune-source/src/main/java/org/opensearch/dataprepper/plugins/source/neptune/converter/NeptuneS3Record.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        }
+
+        String entityType = null;
+        ImmutablePair<String, NeptuneS3RecordPredicate> predicate = null;
+        final PropertygraphData propertygraphData = ((PropertygraphData) record.getData());
+        final String key = propertygraphData.key();
+        if (key.equalsIgnoreCase(PG_ENTITY_ID_KEY)) {
+            entityType = propertygraphData.value().asMap().get("value").asString();
+        } else {
+            predicate = ImmutablePair.of(key, NeptuneS3RecordPredicate.fromPropertGraphData(propertygraphData));
```
Comment (xianjial) [inline data-prepper-plugins/neptune-source/src/main/java/org/opensearch/dataprepper/plugins/source/neptune/converter/NeptuneS3Record.java:71]: nit: typo `fromPropertyGraphData`

**Review by xianjial — APPROVED** [review]

File: data-prepper-plugins/neptune-source/src/main/java/org/opensearch/dataprepper/plugins/source/neptune/converter/NeptuneS3Record.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            predicate = ImmutablePair.of(predicateName, predicateValue);
+        }
+        return NeptuneS3Record
+                .builder()
+                .entityId(stmt.getSubject().stringValue())
+                .documentType(DOCUMENT_TYPE_RDF)
+                .entityType(entityType)
+                .predicate(predicate)
+                .build();
+
```
Comment (halvardataws) [inline data-prepper-plugins/neptune-source/src/main/java/org/opensearch/dataprepper/plugins/source/neptune/converter/NeptuneS3Record.java:111]: NIT: I am wondering if a formatter is needed to be run for the changed files

File: data-prepper-plugins/neptune-source/src/main/java/org/opensearch/dataprepper/plugins/source/neptune/stream/StreamWorker.java [hunk 3]
```diff
@@ -307,7 +294,69 @@ private void bufferWriteAndCheckpointStream() {
...
     }
 
+    /**
+     * If End-to-End acknowledgements are not enabled then we checkpoint every {@link #checkPointIntervalInMs} ms.
+     */
+    private boolean shouldCheckpoint(final long lastCheckpointTime) {
+        return !sourceConfig.isAcknowledgments() && (System.currentTimeMillis() - lastCheckpointTime >= checkPointIntervalInMs);
+    }
+
+    private boolean shouldFlushRecords(final long recordCount) {
```
Comment (halvardataws) [inline data-prepper-plugins/neptune-source/src/main/java/org/opensearch/dataprepper/plugins/source/neptune/stream/StreamWorker.java:304]: NIT: Please add a comment as to the condition for the flushing

**Review by halvardataws — APPROVED** [review]
LGTM, just nitpicks

**Pull request merged**